### PR TITLE
Shrink `grow_amortized()` some more

### PR DESCRIPTION
**Pull request by nnethercote**
More reductions in the amount of LLVM IR generated, to improve build times.

r? @Amanieu

### CODE REVIEW

**Comment by nnethercote** [comment]
@bors rollup=never

Because of the perf effects.

File: src/liballoc/raw_vec.rs [hunk 1]
```diff
@@ -468,20 +449,61 @@ impl<T, A: AllocRef> RawVec<T, A> {
     }
 }
 
+// This function is outside `RawVec` to minimize compile times. See the comment
+// above `RawVec::grow_amortized` for details. (The `A` parameter isn't
```
Comment (lcnr) [inline src/liballoc/raw_vec.rs:453]:
```suggestion
// above `RawVec::grow_amortized` for details. (The `A` parameter isn't used here)
```

**Comment by nnethercote** [comment]
The `check` and `debug` results here are decent, but the `opt` ones are not so good. Hmm.

**Comment by Muirrum** [comment]
@nnethercote This is a triage bump.

**Comment by nnethercote** [comment]
This probably isn't going to lead anywhere useful.

**Comment by nnethercote** [comment]
@SimonSapin: FYI, we discussed more shrinkage of `grow_amortized()` a while back. This was my latest attempt at improving it further. It's fiddly and unpredictable, I think a lot depends on the inlining choices.

**Pull request closed without merging**